sergeant/executor/threaded.py

```diff
@@ -133,6 +133,17 @@ def execute_task(
             self.worker_object.handle_requeue(
                 task=task,
             )
+        except worker.WorkerStop as exception:
```

Comment:
You should streamline the stop signal handling under `except worker.WorkerInterrupt as exception:` on line 147. WorkerStop exception derives from WorkerInterrupt exception class. The second option is to unpack the WorkerInterrupt handling into its parts and leave this function as is but marking the reason for the interruption under `self.interrupt_exception` variable